packages/circuit-ui/components/Modal/Modal.tsx

```diff
@@ -151,15 +151,26 @@ export const Modal: ModalComponent<ModalProps> = ({
   closeButtonLabel,
   className,
   ...props
-}) => (
-  <ClassNames<Theme> key={variant}>
-    {({ css: cssString, cx, theme }) => {
-      // React Modal styles
-      // https://reactcommunity.org/react-modal/styles/classes/
+}) => {
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    process.env.NODE_ENV !== 'test' &&
+    !preventClose &&
+    !closeButtonLabel
+  ) {
+    throw new Error(
+      'The Modal is missing a `closeButtonLabel` prop. This is an accessibility requirement. Pass `preventClose` to hide the close button.',
```

Comment:
I think we should specify here that `closeButtonLabel` should be passed to `setModal` since developers don't render the Modal directly.